lib/transfer.js

```diff
@@ -181,26 +183,41 @@ function _shouldTransferFile(file) {
   return false;
 }
 
+
 /**
  * Request file from URL and upload with filesAdapter
  * @param  {Ibject}   file     the file info object
  */
-function _transferFile(file) {
-  return new Promise(function(resolve, reject) {
+function _transferFile(file) 
+{
+  return new Promise(function(resolve, reject) 
+  {
     if (config.transferTo == 'print') {
       console.log(file.url);
       // Use process.nextTick to avoid max call stack error
       return process.nextTick(resolve);
     }
-    request({ url: file.url, encoding: null }, function(error, response, body) {
-      if (_requestErrorHandler(error, response)) {
-        return reject(error);
-      }
-      config.filesAdapter.createFile(
-        file.newFileName, body, response.headers['content-type']
-      ).then(function() {
-        return _changeDBFileField(file);
-      }).then(resolve, reject);
-    });
+
+	// exists?
+	var localFileName = config.filesAdapter._getLocalFilePath(file.newFileName) ;
+	if ( !fs.existsSync(localFileName) )
```

Comment:
Agreed, as i said, this was something I put together to get it working my end. it will come in useful for others no doubt too. setting the reject to "null" stops the whole process being killed if the request fails, meaning it can carry on. The file exists should take into account s3/gcs too. I've got my parse and files running locally on one machine, with multiple servers running on one mongo db.